crypto/evp/pkey_mac.c

```diff
@@ -95,6 +105,12 @@ static int pkey_mac_copy(EVP_PKEY_CTX *dst, const EVP_PKEY_CTX *src)
     EVP_PKEY_CTX_set_data(dst, dctx);
     dst->keygen_info_count = 0;
 
+    if (sctx->ctx == NULL) {
+        /* This actually means the fetch failed during the init call */
+        EVPerr(0, EVP_R_FETCH_FAILED);
```

Comment:
There is a `ERR_R_INIT_FAIL`...

Reply:
I think EVP_R_FETCH_FAILED is a more useful error. It actually tells you the underlying cause of the problem, and I would think is more likely to lead to the end user figuring out what to do to fix it.

Reply:
I dont have an issue with this..

Reply:
Alternatively there could be a new error reason code created for this - something like EVP_R_MAC_CTX_NOT_FETCHED or EVP_R_MAC_CTX_MISSING

Reply:
I'm fine with any of the three approaches.